src/algebraic_topology/dold_kan/faces.lean

```diff
@@ -0,0 +1,218 @@
+    ring_exp, },
+end
+
+lemma Hσφ_eq_zero {Y : C} {n q : ℕ} (hqn : n<q) {φ : Y ⟶ X _[n+1]}
+  (v : higher_faces_vanish q φ) : φ ≫ (Hσ q).f (n+1) = 0 :=
+begin
+  simp only [Hσ, homotopy.null_homotopic_map'_f (c_mk (n+2) (n+1) rfl) (c_mk (n+1) n rfl)],
+  erw [hσ'_eq_zero hqn (c_mk (n+1) n rfl), comp_zero, zero_add],
+  by_cases hqn' : n+1<q,
+  { rw [hσ'_eq_zero hqn' (c_mk (n+2) (n+1) rfl), zero_comp, comp_zero], },
+  { simp only [hσ'_eq (show n+1=0+q, by linarith) (c_mk (n+2) (n+1) rfl),
+      pow_zero, fin.mk_zero, one_zsmul, eq_to_hom_refl, comp_id],
+    erw chain_complex.of_d,
+    simp only [alternating_face_map_complex.obj_d, comp_sum],
+    rw [← fin.sum_congr' _ (show 2+(n+1)=n+1+2, by linarith), fin.sum_trunc],
+    { simp only [fin.sum_univ_cast_succ, fin.sum_univ_zero, zero_add, fin.last,
+        fin.cast_le_mk, fin.cast_mk, fin.cast_succ_mk],
+      simp only [fin.mk_zero, fin.coe_zero, pow_zero, one_zsmul, fin.mk_one,
+        fin.coe_one, pow_one, neg_smul, comp_neg],
+      erw [δ_comp_σ_self, δ_comp_σ_succ, add_right_neg], },
+    { intro j,
+      simp only [comp_zsmul],
+      convert zsmul_zero _,
+      have h : fin.cast (by rw add_comm 2) (fin.nat_add 2 j) = j.succ.succ,
+      { ext, simp only [add_comm 2, fin.coe_cast, fin.coe_nat_add, fin.coe_succ], },
+      rw [h, ← fin.cast_succ_zero, δ_comp_σ_of_gt X], swap,
+      { exact fin.succ_pos j, },
+      simp only [← assoc, v j (by linarith), zero_comp], }, },
+end
+
+lemma higher_faces_vanish_induction {Y : C} {n q : ℕ} {φ : Y ⟶ X _[n+1]}
+  (v : higher_faces_vanish q φ) : higher_faces_vanish (q+1) (φ ≫ (𝟙 _ + Hσ q).f (n+1)) :=
+begin
+  intros j hj₁,
+  simp only [add_comp, comp_add, homological_complex.add_f_apply, homological_complex.id_f],
+  erw comp_id,
+  -- when n < q, the result follows immediately from the assumption
+  by_cases hqn : n<q,
+  { rw [Hσφ_eq_zero hqn v, zero_comp, add_zero, v j (by linarith)], },
+  -- we now assume that n≥q, and write n=a+q
+  cases nat.le.dest (not_lt.mp hqn) with a ha,
+  rw [Hσφ_eq_neg_σδφ (show n=a+q, by linarith) v,
+    neg_comp, add_neg_eq_zero, assoc, assoc],
+  cases n with m hm,
+  -- the boundary case n=0
+  { simp only [nat.eq_zero_of_add_eq_zero_left ha, fin.eq_zero j,
+      fin.mk_zero, fin.mk_one],
+    erw [δ_comp_σ_succ],
+    simp only [fin.succ_zero_eq_one, comp_id], },
+  -- in the other case, we need to write n as m+1
+  -- then, we first consider the particular case j = a
+  by_cases hj₂ : a = (j : ℕ),
+  { simp only [hj₂, fin.eta, δ_comp_σ_succ, comp_id],
+    congr,
+    ext,
+    simp only [fin.coe_succ, fin.coe_mk], },
+  -- now, we assume j ≠ a (i.e. a < j)
+  have haj : a<j := (ne.le_iff_lt hj₂).mp (by linarith),
+  have hj₃ := j.is_lt,
+  have ham : a≤m,
+  { by_contradiction,
+    rw [not_le, ← nat.succ_le_iff] at h,
+    linarith, },
+  have ineq₁ : (fin.cast_succ (⟨a, nat.lt_succ_iff.mpr ham⟩ : fin(m+1)) < j),
+  { rw fin.lt_iff_coe_lt_coe, exact haj, },
+  erw δ_comp_σ_of_gt X ineq₁,
```

Comment:
I wonder if this `erw` is an indication that we need an alternative version of `δ_comp_σ_of_gt` that can be used directly here?

Reply:
Overall, I have struggled when using the simplicial relations. Sometimes, I had to do some rewrites into the simplicial equations before I could (e)rw them, but here I have quite happy that I could `erw` it.